pinot-spi/src/main/java/org/apache/pinot/spi/recordenricher/RecordEnricherFactoryInterface.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.recordenricher;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+
+
+public interface RecordEnricherFactoryInterface {
+  String getEnricherType();
+  RecordEnricher createEnricher(JsonNode enricherProps) throws IOException;
+  void validateEnrichmentConfig(JsonNode enricherProps, boolean disableGroovy);
```

Comment:
I don't like to have that `disableGroovy` param here. If in future we add other script languages that can be disabled, we would need to add more paramentres. A more robust design would be to have here either some contextual object (a Context that inside contains whether groovy is disabled or not) or even better the PinotConfig itself.

Reply:
I created a new class to pass the constraints to the enricher for validation